src/mame/drivers/galaxian_rockclim.cpp

```diff
@@ -0,0 +1,294 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#include "emu.h"
+#include "includes/galaxian.h"
+
+class galaxian_rockclim_state : public galaxian_state
+{
+public:
+	galaxian_rockclim_state(const machine_config &mconfig, device_type type, const char *tag)
+		: galaxian_state(mconfig, type, tag)
+		, m_rockclim_videoram(*this,"rockclim_vram")
+		, m_gfxdecode2(*this,"gfxdecode2")
+		, m_palette2(*this,"palette2")
+	{ }
+
+	void rockclim(machine_config &config);
+
+	void init_rockclim();
+
+private:
+	virtual void video_start() override;
+
+	void rockclim_map(address_map &map);
+
+	void rockclim_extend_sprite_info(const uint8_t *base, uint8_t *sx, uint8_t *sy, uint8_t *flipx, uint8_t *flipy, uint16_t *code, uint8_t *color);
+	void rockclim_videoram_w(offs_t offset, uint8_t data);
+	void rockclim_scroll_w(offs_t offset, uint8_t data);
+	uint8_t rockclim_videoram_r(offs_t offset);
+	optional_shared_ptr<uint8_t> m_rockclim_videoram;
+
+	void rockclim_palette(palette_device &palette) const;
+
+	required_device<gfxdecode_device> m_gfxdecode2;
+	required_device<palette_device> m_palette2;
+
+	TILE_GET_INFO_MEMBER(rockclim_get_tile_info);
+	tilemap_t *m_rockclim_tilemap;
+	uint16_t m_rockclim_v;
+	uint16_t m_rockclim_h;
+
+	void rockclim_draw_background(bitmap_rgb32 &bitmap, const rectangle &cliprect);
+};
+
+void galaxian_rockclim_state::rockclim_draw_background(bitmap_rgb32 &bitmap, const rectangle &cliprect)
+{
+	m_rockclim_tilemap->draw(*m_screen, bitmap, cliprect, 0,0);
+}
+
+void galaxian_rockclim_state::rockclim_extend_sprite_info(const uint8_t *base, uint8_t *sx, uint8_t *sy, uint8_t *flipx, uint8_t *flipy, uint16_t *code, uint8_t *color)
+{
+	if ((*code & 0x30) == 0x20)
+	{
+		if (m_gfxbank[2] & 1)
+		{
+			int bank = (((m_gfxbank[0] & 1) << 5) | ((m_gfxbank[1] & 1) << 4));
+			*code = (0x40 + bank) | (*code & 0x0f);
+		}
+	}
+}
+
+uint8_t galaxian_rockclim_state::rockclim_videoram_r(offs_t offset)
+{
+	return m_rockclim_videoram[offset];
+}
+
+void galaxian_rockclim_state::rockclim_videoram_w(offs_t offset, uint8_t data)
+{
+	m_rockclim_videoram[offset] = data;
+	m_rockclim_tilemap->mark_tile_dirty(offset);
+}
+
+void galaxian_rockclim_state::rockclim_scroll_w(offs_t offset, uint8_t data)
+{
+	switch (offset & 3)
+	{
+	case 0: m_rockclim_h = (m_rockclim_h & 0xff00) | data; m_rockclim_tilemap->set_scrollx(0, m_rockclim_h * GALAXIAN_XSCALE); break;
+	case 1: m_rockclim_h = (m_rockclim_h & 0x00ff) | (data << 8); m_rockclim_tilemap->set_scrollx(0, m_rockclim_h * GALAXIAN_XSCALE); break;
```

Comment:
You should be using `m_x_scale` rather than the constant here in case it’s changed by machine configuration for whatever reason.